Testing: rename `NO_XDIST` flag to `RUN_XDIST`

mgajek-cern opened the pull request:
Addresses #7010

mgajek-cern committed:
chore(testing): rename NO_XDIST flag to RUN_XDIST

rdimaio commented on tools/pytest.sh:
```diff
@@ -30,28 +30,28 @@ fi
   # no parallel tests on sqlite, because of random "sqlite3.OperationalError: database is locked"
   echo "Disabling parallel testing for sqlite"
-  NO_XDIST="True"
+  RUN_XDIST="True"
```
You should also flip the bool logic: if `NO_XDIST` is true, we don't run XDIST. But if `RUN_XDIST` is true, we run XDIST. These semantically have opposite meaning, so every instance should be flipped accordingly: e.g. `NO_XDIST` was `False` by default, so `RUN_XDIST` should be `True` by default, and so on.

mgajek-cern replied:
Addressed in subsequent commit

mgajek-cern committed:
fix(Testing): correct RUN_XDIST boolean logic and add proper xdist availability check

- Flip RUN_XDIST default from False to True (run xdist by default)
- Fix database-specific conditions to set RUN_XDIST=False (disable xdist)
- Add proper xdist availability check that only disables when import fails
- Improve error handling and user feedback for missing xdist package

Previously the logic was inverted: RUN_XDIST=False would enable xdist,
and the xdist availability check would disable xdist even when available.

Fixes the semantic meaning to match variable name: RUN_XDIST=True means
"run with xdist enabled".

mgajek-cern commented:
Closed due to incorrect branch name. See updated PR: https://github.com/rucio/rucio/pull/7916
